### PR TITLE
GraphQLApi support for `sam deploy`

### PR DESCRIPTION
**Pull request by ssenchenko**
#### Why is this change necessary?
Similar to [this PR](https://github.com/aws/aws-sam-cli/pull/5250) for `sam package`, `sam build` has to support updating relative code paths for `GraphQLApi` resource with multiple properties (in different sub-dictionaries) with the same name `CodeUri`.

I also did a bit of refactoring, moving the function which traverses `GraphQLApi` into `lib/utils` out of the `GraphQLApiCodeResource` class.

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [x] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/lib/package/packageable_resources.py [hunk 1]
```diff
@@ -599,7 +600,7 @@ def get_property_value(self, resource_dict):
 
 class GraphQLApiSchemaResource(ResourceZip):
     RESOURCE_TYPE = AWS_SERVERLESS_GRAPHQLAPI
-    PROPERTY_NAME = RESOURCES_WITH_LOCAL_PATHS[RESOURCE_TYPE][0]
+    PROPERTY_NAME = graphql_api.SCHEMA_ARTIFACT_PROPERTY
```
Comment (mildaniel) [inline samcli/lib/package/packageable_resources.py:603]: nit: Can we import anything we need from `graphql_api` directly instead of importing the module?

**Review by mildaniel — COMMENTED** [review]
Thanks Slava, LGTM! Can we add an integration test case, in this or separate PR?

**Review by mildaniel — APPROVED** [review]

File: samcli/lib/package/packageable_resources.py (hunk 1 above)
Comment (ssenchenko) [reply samcli/lib/package/packageable_resources.py:603]: Is the style I used is against SAM CLI practice? I import everything from that file anyway. But the main reason, I think that `graphql_api.SCHEMA_ARTIFACT_PROPERTY` and `graphql_api.find_all_paths_and_values(self.PROPERTY_NAME, resource_dict)` is easier to read then simply `SCHEMA_ARTIFACT_PROPERTY` and `find_all_paths_and_values` which show less "belonging". But if it's against the style, I can remove it. It's up to you

Comment (hawflau) [reply samcli/lib/package/packageable_resources.py:603]: I think it's fine since `graphql_api` is not a huge module. And +1 for the point of "belonging".

**Review by hawflau — APPROVED** [review]

**Pull request merged**